cdap-api/src/main/java/co/cask/cdap/api/lineage/LineageRecorder.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.lineage;
+
+import co.cask.cdap.api.lineage.operation.Operation;
+import co.cask.cdap.api.lineage.operation.Read;
+import co.cask.cdap.api.lineage.operation.Write;
+
+import java.util.Collection;
+
+/**
+ * Interface for recording the lineage information.
+ */
+public interface LineageRecorder {
+  /**
+   * Record the lineage operations.
```

Comment:
The way this is worded is sounds that this can be called only once? (Record "the" lineage operations). Or can it be called multiple times by the same program?

Reply:
There are 2 options:
1. When this method is called, the collection of operation should represent the COMPLETE collection (from Read to Write), meaning if it is called multiple times from program we will overwrite the previous collection. This is what I was thinking and will make javadoc more clear).
2. We allow calling `record` multiple times by program and in the end collection of operations would be all the operations provided by record method. In this case I think we should also expose the 
`void record(Operation operation);`

Let me know what do you think.

Reply:
It is fine in my opinion if you can call it only once. But I think it throw an exception if called again. Otherwise people might not notice that they are dropping previously recorded operations.